crates/apub/src/activities/create_or_update/comment.rs

```diff
@@ -49,10 +50,19 @@ impl SendActivity for CreateComment {
   type Response = CommentResponse;
 
   async fn send_activity(
-    _request: &Self,
+    request: &Self,
     response: &Self::Response,
     context: &LemmyContext,
   ) -> Result<(), LemmyError> {
+    let local_site = LocalSite::read(context.pool()).await?;
+    let local_user_view =
+      get_local_user_view_from_jwt(&request.auth, context.pool(), context.secret()).await?;
+    if local_site.registration_mode.require_approval()
+      && !local_user_view.local_user.accepted_application
+    {
+      return Ok(());
```

Comment:
Shouldn't this be an error? Seems like this should be further up a chain somewhere, and never try to send the activity in the first place.

I leave it to you tho, if this works then its good.